### DISCUSSION TITLE
Peacock -->vtk=7.1.1.opengl - PackagesNotFoundError

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#16761] [Q&A General] [answered]
Dear all,
I recently switched to a new HPC environment and was trying to configure peacock there. I went though whole process of installing miniconda (https://mooseframework.inl.gov/getting_started/installation/manual_miniconda.html) and other required packages, before invoking peacock and the following rendering issue.
As a cure, i tried to installl vtk version 7.1.1 in a new conda environment, but returned with an error saying that the concerned package is not available in idaholab channel. Any advise on fixing this issue would be highly appreciated.
Kind regards,
Arun

2021-01-20 18:22:52.934 (   3.566s) [        8A2DC380]vtkOpenGLRenderWindow.c:582    ERR| vtkXOpenGLRenderWindow (0x5628f55060f0): Unable to find a valid OpenGL 3.2 or later implementation. Please update your video card driver to the latest version. If you are using Mesa please make sure you have version 11.2 or later and make sure your driver in Mesa supports OpenGL 3.2 such as llvmpipe or openswr. If you are on windows and using Microsoft remote desktop note that it only supports OpenGL 3.2 with nvidia quadro cards. You can use other remoting software such as nomachine to avoid this issue.
2021-01-20 18:22:52.936 (   3.568s) [        8A2DC380]     vtkOpenGLState.cxx:1380  WARN| Hardware does not support the number of textures defined.
2021-01-20 18:22:52.937 (   3.569s) [        8A2DC380]     vtkOpenGLState.cxx:1380  WARN| Hardware does not support the number of textures defined.
ERROR: In /vtk/src/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx, line 582
vtkXOpenGLRenderWindow (0x5628f55060f0): Unable to find a valid OpenGL 3.2 or later implementation. Please update your video card driver to the latest version. If you are using Mesa please make sure you have version 11.2 or later and make sure your driver in Mesa supports OpenGL 3.2 such as llvmpipe or openswr. If you are on windows and using Microsoft remote desktop note that it only supports OpenGL 3.2 with nvidia quadro cards. You can use other remoting software such as nomachine to avoid this issue.

Couldn't show a VTK window.
This is most likely caused by outdated Mesa drivers.
You can try upgrading to Mesa version 10.6.5 or later or update your video card driver.
If you are running in a virtual machine, you can try turning OFF 3D acceleration.

You can also try to use an alternative VTK package in a conda environment.
You can create a new conda environment by doing (with the miniconda module loaded):

        conda create --clone root -n <env_name>
        source activate <env_name>
        conda install -c idaholab vtk=7.1.1.opengl

Then you can try running peacock again.
Note that this VTK package may have various rendering problems and some peacock tests will fail.
(Arun_conda_env) [ir-bala2@login-e-13 peacock]$ source activate Arun_conda_env
(Arun_conda_env) [ir-bala2@login-e-13 peacock]$ conda install -c idaholab vtk=7.1.1.opengl
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - vtk=7.1.1.opengl

Current channels:

  - https://conda.anaconda.org/idaholab/linux-64
  - https://conda.anaconda.org/idaholab/noarch
  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

**Comment by aeslaughter** [comment, marked as answer]
VTK8.2 from conda-forge should work. @milljm Correct me if I am wrong, but we do not have the VTK7 in the idaholab channel any more. 

@abarun22 Did you see VTK7.1 in the directions somewhere?

**Reply by abarun22** [reply]
Hi Andrew, I did find other packages for VTK through pip installations in conda environment. But none of them are opengl compatible, which would bring back the same rendering problem. Versions 7.1.1 in conda-forge clashes with the native python version (3.7.4) and are not installing. The subsequent >8.0.0 versions contains many other installations which overwrite my python and pyqt5 installations and pose much more difficulties. I remember in my previous HPC i used the one (7.1.1) from idaholab which install VTK ok and enabled peacock to open successfully. It would be helpful if you could point me to vtk packages that are render compatible. The following are available packages in various channels in my conda environment.

(Arun_conda_env) [ir-bala2@login-e-14 peacock]$ conda search -c conda-forge | grep -i 'vtk'
pyevtk                         1.1.1    pyh24bf2e0_0  conda-forge
pyevtk                         1.1.2    pyh24bf2e0_0  conda-forge
pyvtk                         0.5.18          py27_0  conda-forge
pyvtk                         0.5.18          py35_0  conda-forge
pyvtk                         0.5.18          py36_0  conda-forge
vtk                            7.0.0          py27_0  conda-forge
vtk                            7.0.0          py27_1  conda-forge
vtk                            7.0.0          py27_2  conda-forge
vtk                            7.0.0          py35_0  conda-forge
vtk                            7.0.0          py35_1  conda-forge
vtk                            7.0.0          py35_2  conda-forge
vtk                            7.1.0          py27_0  conda-forge
vtk                            7.1.0          py27_1  conda-forge
vtk                            7.1.0          py27_2  conda-forge
vtk                            7.1.0          py27_3  conda-forge
vtk                            7.1.0          py27_4  conda-forge
vtk                            7.1.0          py27_5  conda-forge
vtk                            7.1.0          py35_0  conda-forge
vtk                            7.1.0          py35_1  conda-forge
vtk                            7.1.0          py35_2  conda-forge
vtk                            7.1.0          py35_3  conda-forge
vtk                            7.1.0          py35_4  conda-forge
vtk                            7.1.0          py36_3  conda-forge
vtk                            7.1.0          py36_4  conda-forge
vtk                            7.1.1          py27_0  conda-forge
vtk                            7.1.1          py27_1  conda-forge
vtk                            7.1.1        py27_201  conda-forge
vtk                            7.1.1        py27_202  conda-forge
vtk                            7.1.1  py27_mesalib_1  conda-forge
vtk                            7.1.1  py27_mesalib_2  conda-forge
vtk                            7.1.1          py35_0  conda-forge
vtk                            7.1.1          py35_1  conda-forge
vtk                            7.1.1        py35_201  conda-forge
vtk                            7.1.1        py35_202  conda-forge
vtk                            7.1.1  py35_mesalib_1  conda-forge
vtk                            7.1.1  py35_mesalib_2  conda-forge
vtk                            7.1.1          py36_0  conda-forge
vtk                            7.1.1          py36_1  conda-forge
vtk                            7.1.1        py36_201  conda-forge
vtk                            7.1.1        py36_202  conda-forge
vtk                            7.1.1  py36_mesalib_1  conda-forge
vtk                            7.1.1  py36_mesalib_2  conda-forge
vtk                            8.0.1        py27_200  conda-forge
vtk                            8.0.1  py27_mesalib_0  conda-forge
vtk                            8.0.1        py35_200  conda-forge
vtk                            8.0.1  py35_mesalib_0  conda-forge
vtk                            8.0.1        py36_200  conda-forge
vtk                            8.0.1  py36_mesalib_0  conda-forge
vtk                            8.1.0        py27_200  conda-forge
vtk                            8.1.0        py27_201  conda-forge
vtk                            8.1.0        py27_202  conda-forge
vtk                            8.1.0  py27_mesalib_0  conda-forge
vtk                            8.1.0  py27_mesalib_1  conda-forge
vtk                            8.1.0  py27_mesalib_2  conda-forge
vtk                            8.1.0 py27_mesalibh9c317cf_4  conda-forge
vtk                            8.1.0 py27_mesalibh9c317cf_5  conda-forge
vtk                            8.1.0 py27_mesalibhc26a1c2_3  conda-forge
vtk                            8.1.0 py27_mesalibhd4c0a90_2  conda-forge
vtk                            8.1.0 py27h3f2a929_203  conda-forge
vtk                            8.1.0 py27h9686630_201  pkgs/main
vtk                            8.1.0 py27h969b564_201  pkgs/main
vtk                            8.1.0 py27h99a48a5_204  conda-forge
vtk                            8.1.0 py27h99a48a5_205  conda-forge
vtk                            8.1.0 py27he461554_202  conda-forge
vtk                            8.1.0        py35_200  conda-forge
vtk                            8.1.0        py35_201  conda-forge
vtk                            8.1.0        py35_202  conda-forge
vtk                            8.1.0  py35_mesalib_0  conda-forge
vtk                            8.1.0  py35_mesalib_1  conda-forge
vtk                            8.1.0  py35_mesalib_2  conda-forge
vtk                            8.1.0 py35_mesalibh9c317cf_4  conda-forge
vtk                            8.1.0 py35_mesalibh9c317cf_5  conda-forge
vtk                            8.1.0 py35_mesalibhc26a1c2_3  conda-forge
vtk                            8.1.0 py35_mesalibhd4c0a90_2  conda-forge
vtk                            8.1.0 py35h3f2a929_203  conda-forge
vtk                            8.1.0 py35h9686630_201  pkgs/main
vtk                            8.1.0 py35h969b564_201  pkgs/main
vtk                            8.1.0 py35h99a48a5_204  conda-forge
vtk                            8.1.0 py35h99a48a5_205  conda-forge
vtk                            8.1.0 py35he461554_202  conda-forge
vtk                            8.1.0        py36_200  conda-forge
vtk                            8.1.0        py36_201  conda-forge
vtk                            8.1.0        py36_202  conda-forge
vtk                            8.1.0  py36_mesalib_0  conda-forge
vtk                            8.1.0  py36_mesalib_1  conda-forge
vtk                            8.1.0  py36_mesalib_2  conda-forge
vtk                            8.1.0 py36_mesalibh9c317cf_4  conda-forge
vtk                            8.1.0 py36_mesalibh9c317cf_5  conda-forge
vtk                            8.1.0 py36_mesalibhc26a1c2_3  conda-forge
vtk                            8.1.0 py36_mesalibhd4c0a90_2  conda-forge
vtk                            8.1.0 py36h3f2a929_203  conda-forge
vtk                            8.1.0 py36h9686630_201  pkgs/main
vtk                            8.1.0 py36h969b564_201  pkgs/main
vtk                            8.1.0 py36h99a48a5_204  conda-forge
vtk                            8.1.0 py36h99a48a5_205  conda-forge
vtk                            8.1.0 py36he461554_202  conda-forge
vtk                            8.1.1 py27_mesalibh9c317cf_0  conda-forge
vtk                            8.1.1 py27_mesalibhed38e9a_4  conda-forge
vtk                            8.1.1 py27_mesalibhf80e44b_1  conda-forge
vtk                            8.1.1 py27_mesalibhf884bf3_2  conda-forge
vtk                            8.1.1 py27_mesalibhf884bf3_3  conda-forge
vtk                            8.1.1 py27h4f1983a_204  pkgs/main
vtk                            8.1.1 py27h8649a91_204  conda-forge
vtk                            8.1.1 py27h881b6ac_204  pkgs/main
vtk                            8.1.1 py27h99a48a5_200  conda-forge
vtk                            8.1.1 py27hdceb404_201  conda-forge
vtk                            8.1.1 py27hdfe3907_202  conda-forge
vtk                            8.1.1 py27hdfe3907_203  conda-forge
vtk                            8.1.1 py35_mesalibh9c317cf_0  conda-forge
vtk                            8.1.1 py35_mesalibhed38e9a_4  conda-forge
vtk                            8.1.1 py35_mesalibhf80e44b_1  conda-forge
vtk                            8.1.1 py35_mesalibhf884bf3_2  conda-forge
vtk                            8.1.1 py35_mesalibhf884bf3_3  conda-forge
vtk                            8.1.1 py35h8649a91_204  conda-forge
vtk                            8.1.1 py35h99a48a5_200  conda-forge
vtk                            8.1.1 py35hdceb404_201  conda-forge
vtk                            8.1.1 py35hdfe3907_202  conda-forge
vtk                            8.1.1 py35hdfe3907_203  conda-forge
vtk                            8.1.1 py36_mesalibh9c317cf_0  conda-forge
vtk                            8.1.1 py36_mesalibhed38e9a_4  conda-forge
vtk                            8.1.1 py36_mesalibhf80e44b_1  conda-forge
vtk                            8.1.1 py36_mesalibhf884bf3_2  conda-forge
vtk                            8.1.1 py36_mesalibhf884bf3_3  conda-forge
vtk                            8.1.1 py36h4f1983a_204  pkgs/main
vtk                            8.1.1 py36h8649a91_204  conda-forge
vtk                            8.1.1 py36h881b6ac_204  pkgs/main
vtk                            8.1.1 py36h99a48a5_200  conda-forge
vtk                            8.1.1 py36hdceb404_201  conda-forge
vtk                            8.1.1 py36hdfe3907_202  conda-forge
vtk                            8.1.1 py36hdfe3907_203  conda-forge
vtk                            8.1.2 py27h8cc7ba1_1200  conda-forge
vtk                            8.1.2 py36h8cc7ba1_1200  conda-forge
vtk                            8.1.2 py37h8cc7ba1_1200  conda-forge
vtk                            8.2.0 py27h01fbe97_209  conda-forge
vtk                            8.2.0 py27h06adf98_210  conda-forge
vtk                            8.2.0 py27h3196ccb_208  conda-forge
vtk                            8.2.0 py27h4804ed4_200  conda-forge
vtk                            8.2.0 py27h4a8345c_204  conda-forge
vtk                            8.2.0 py27h9286700_207  conda-forge
vtk                            8.2.0 py27ha8e561a_201  conda-forge
vtk                            8.2.0 py27ha8e561a_202  conda-forge
vtk                            8.2.0 py27haa4764d_200  pkgs/main
vtk                            8.2.0 py27hd1c59f8_205  conda-forge
vtk                            8.2.0 py27hd1c59f8_206  conda-forge
vtk                            8.2.0 py27hfdee58b_203  conda-forge
vtk                            8.2.0 py36h00504ea_210  conda-forge
vtk                            8.2.0 py36h00504ea_211  conda-forge
vtk                            8.2.0 py36h06fa586_215  conda-forge
vtk                            8.2.0 py36h06fa586_216  conda-forge
vtk                            8.2.0 py36h06fa586_217  conda-forge
vtk                            8.2.0 py36h06fa586_218  conda-forge
vtk                            8.2.0 py36h3196ccb_208  conda-forge
vtk                            8.2.0 py36h4804ed4_200  conda-forge
vtk                            8.2.0 py36h4a8345c_204  conda-forge
vtk                            8.2.0 py36h5457cd2_212  conda-forge
vtk                            8.2.0 py36h5457cd2_213  conda-forge
vtk                            8.2.0 py36h863d835_214  conda-forge
vtk                            8.2.0 py36h9286700_207  conda-forge
vtk                            8.2.0 py36ha8e561a_201  conda-forge
vtk                            8.2.0 py36ha8e561a_202  conda-forge
vtk                            8.2.0 py36haa4764d_200  pkgs/main
vtk                            8.2.0 py36hd1c59f8_205  conda-forge
vtk                            8.2.0 py36hd1c59f8_206  conda-forge
vtk                            8.2.0 py36hfd09df2_209  conda-forge
vtk                            8.2.0 py36hfdee58b_203  conda-forge
vtk                            8.2.0 py37h2bd422c_215  conda-forge
vtk                            8.2.0 py37h2bd422c_216  conda-forge
vtk                            8.2.0 py37h2bd422c_217  conda-forge
vtk                            8.2.0 py37h2bd422c_218  conda-forge
vtk                            8.2.0 py37h3196ccb_208  conda-forge
vtk                            8.2.0 py37h351a6da_214  conda-forge
vtk                            8.2.0 py37h4804ed4_200  conda-forge
vtk                            8.2.0 py37h4a8345c_204  conda-forge
vtk                            8.2.0 py37h55fc74f_212  conda-forge
vtk                            8.2.0 py37h55fc74f_213  conda-forge
vtk                            8.2.0 py37h9286700_207  conda-forge
vtk                            8.2.0 py37ha8e561a_201  conda-forge
vtk                            8.2.0 py37ha8e561a_202  conda-forge
vtk                            8.2.0 py37haa4764d_200  pkgs/main
vtk                            8.2.0 py37hbd17b0a_210  conda-forge
vtk                            8.2.0 py37hbd17b0a_211  conda-forge
vtk                            8.2.0 py37hc9f1d74_209  conda-forge
vtk                            8.2.0 py37hd1c59f8_205  conda-forge
vtk                            8.2.0 py37hd1c59f8_206  conda-forge
vtk                            8.2.0 py37hfdee58b_203  conda-forge
vtk                            8.2.0 py38h1e82295_210  conda-forge
vtk                            8.2.0 py38h1e82295_211  conda-forge
vtk                            8.2.0 py38h3196ccb_208  conda-forge
vtk                            8.2.0 py38h4a8345c_204  conda-forge
vtk                            8.2.0 py38h667666d_212  conda-forge
vtk                            8.2.0 py38h667666d_213  conda-forge
vtk                            8.2.0 py38h9286700_207  conda-forge
vtk                            8.2.0 py38hb756a82_209  conda-forge
vtk                            8.2.0 py38hd1c59f8_205  conda-forge
vtk                            8.2.0 py38hd1c59f8_206  conda-forge
vtk                            8.2.0 py38hd6830a8_214  conda-forge
vtk                            8.2.0 py38hf2e56f5_215  conda-forge
vtk                            8.2.0 py38hf2e56f5_216  conda-forge
vtk                            8.2.0 py38hf2e56f5_217  conda-forge
vtk                            8.2.0 py38hf2e56f5_218  conda-forge
vtk                            9.0.0 no_osmesa_py36h1d9199b_100  conda-forge
vtk                            9.0.0 no_osmesa_py36h1d9199b_101  conda-forge
vtk                            9.0.0 no_osmesa_py36h77cafce_102  conda-forge
vtk                            9.0.0 no_osmesa_py37h51040a7_100  conda-forge
vtk                            9.0.0 no_osmesa_py37h51040a7_101  conda-forge
vtk                            9.0.0 no_osmesa_py37h53e2506_102  conda-forge
vtk                            9.0.0 no_osmesa_py38hc86227b_100  conda-forge
vtk                            9.0.0 no_osmesa_py38hc86227b_101  conda-forge
vtk                            9.0.1 no_osmesa_py36h0d9dc2d_102  conda-forge
vtk                            9.0.1 no_osmesa_py36h4e529f2_101  conda-forge
vtk                            9.0.1 no_osmesa_py36h4e529f2_102  conda-forge
vtk                            9.0.1 no_osmesa_py36h77cafce_100  conda-forge
vtk                            9.0.1 no_osmesa_py37h53e2506_100  conda-forge
vtk                            9.0.1 no_osmesa_py37h8af0cc8_101  conda-forge
vtk                            9.0.1 no_osmesa_py37h8af0cc8_102  conda-forge
vtk                            9.0.1 no_osmesa_py37haba06c1_102  conda-forge
vtk                            9.0.1 no_osmesa_py38h31de780_101  conda-forge
vtk                            9.0.1 no_osmesa_py38h31de780_102  conda-forge
vtk                            9.0.1 no_osmesa_py38h87d9d97_102  conda-forge
vtk                            9.0.1 no_osmesa_py38he0bb71e_100  conda-forge
vtk                            9.0.1 no_osmesa_py39h9ad716d_102  conda-forge
vtk                            9.0.1 no_osmesa_py39hd84e346_101  conda-forge
vtk                            9.0.1 no_osmesa_py39hd84e346_102  conda-forge

(Arun_conda_env) [ir-bala2@login-e-14 peacock]$ conda search vtk
Loading channels: done
# Name                       Version           Build  Channel
vtk                            8.1.0 py27h9686630_201  pkgs/main
vtk                            8.1.0 py27h969b564_201  pkgs/main
vtk                            8.1.0 py35h9686630_201  pkgs/main
vtk                            8.1.0 py35h969b564_201  pkgs/main
vtk                            8.1.0 py36h9686630_201  pkgs/main
vtk                            8.1.0 py36h969b564_201  pkgs/main
vtk                            8.1.1 py27h4f1983a_204  pkgs/main
vtk                            8.1.1 py27h881b6ac_204  pkgs/main
vtk                            8.1.1 py36h4f1983a_204  pkgs/main
vtk                            8.1.1 py36h881b6ac_204  pkgs/main
vtk                            8.2.0 py27haa4764d_200  pkgs/main
vtk                            8.2.0 py36haa4764d_200  pkgs/main
vtk                            8.2.0 py37haa4764d_200  pkgs/main

Kind regards,
Arun

**Reply by aeslaughter** [reply]
You might try installing mesalib with VTK:
```
conda create -n vtk-mesa mesalib vtk
```

**Reply by milljm** [reply]
@aeslaughter Correct. We use conda-forge's VTK. The only VTK we distribute ourselves is for use by libMesh's VTK support (moose-libmesh-vtk).

@abarun22 Conda does not distribute OpenGL, or mesa (not the mesa we're all thinking. Mesa3d is the product of that error). Conda's VTK simply uses whats available on the machine. If the version available on the machine is too old, you may see a message like that. You are correct in trying to get an older version.

Unfortunately after digging a little, I'm afraid VTK 7.1.1 was only ever built with PyQt 4 (while Peacock requires 5). The errors you are receiving I believe are due to that machine needing a newer implementation of OpenGL and/or Mesa just as the VTK error suggests...
